fecgov/fec-cms
# AO Comment deadlines display

## Summary (required)

- Resolves #5400
- Adds view logic and template changes to display AO comment deadlines extracted from document descriptions
- Update `How to Comment` link to point to anchor

### Required reviewers

UX/Frontend

## Impacted areas of the application
  AO Landing page
	modified:   legal/templates/legal-advisory-opinions-landing.jinja
	modified:   legal/views.py

## Screenshots

<img width="906" alt="Screen Shot 2022-10-24 at 9 57 09 PM" src="https://user-images.githubusercontent.com/5572856/197664161-a9cd6725-9ca2-4c8e-99af-0d806a698977.png">


## Related PRs

## How to test

- Checkout and run branch
- Since there are currently no AOs with pending comment dates, there is sample data dict in the view called `pending_aos_sample` . This is the data output from the exact same call to the api that the `pending_aos var`  makes to api_caller with the document descriptions edited allow some AOs comment dates to show on page. (See comment on line 399)
- Edit the doc description "Comments due by..."  section to test reliability of the regex and view logic 
- Edit the doc description "Comments due by..." section to see what typos are forgiven and which are not(case insensitivity and extra spaces, 1-digit month without leading zero).
- To cover all edge cases, `AO 2022-23` in the sample data has two non-draft documents with comment deadlines.  Although Jason confirmed that this would not be a normal occurrence, I would wager on it happening some time in the future, so the last document added, is the one used to extract the deadline. 
- If it's helpful to see what the pending_aos' json dicts look like after the view logic adds the `comment_deadline` to it ,  you can uncomment lines 86-89 of the template. This renders a formatted version of the resulting `pending_aos` sample data to the page. (See comment on line 407 of the view for explanation.)
- You could also test the regex, saved  here https://regex101.com/r/6ymj4l/1

**Note:**
The temporary testing code is the only way I could think of to make it easy for devs to test and have confidence in the logic given that there are no pending dates in live data at time of submitting the PR.  
-  [x] Once this WIP is tested, I can push a cleaner PR that represents the final version of the view and template
